islandora_object_lock_length/islandora_object_lock_length.module

```diff
@@ -80,22 +124,35 @@ function islandora_object_lock_length_manage_lock_form(array $form, array &$form
  */
 function islandora_object_lock_length_manage_lock_form_submit(array $form, array &$form_state) {
   module_load_include('inc', 'islandora_object_lock_length', 'includes/utilities');
-  // Get Collection lock duration if it is already set.
-  $collection_lock_length = islandora_object_lock_length_get_collection_lock($form_state['storage']['pid']);
-  $default_lock_length = variable_get('islandora_object_lock_duration', 30);
+  if ($form_state['triggering_element']['#parents'] == array('submit')) {
+    // Get Collection lock duration if it is already set.
+    $collection_lock_length = islandora_object_lock_length_get_collection_lock($form_state['storage']['pid']);
+    $default_lock_length = variable_get('islandora_object_lock_duration', 30);
 
-  // No need to add the lock value if it is the same as the global default lock.
-  if ($form['duration']['#value'] != $default_lock_length) {
-    // Delete old value, if it exists.
-    if (isset($collection_lock_length->duration)) {
-      islandora_object_lock_length_remove_collection_lock($form_state['storage']['pid']);
+    // No need to add the lock value if it is the same as the global default
+    // lock. See if the choice and default values match before we do anything.
+    $duration = $form_state['values']['duration'];
+    $choice = $form_state['values']['choice'];
+    if (($choice == 'minutes' && $duration != $default_lock_length) || ($choice == 'indefinite' && $default_lock_length != FALSE)) {
+      // Delete old value, if it exists.
+      if (isset($collection_lock_length->duration)) {
+        islandora_object_lock_length_remove_collection_lock($form_state['storage']['pid']);
+      }
+      // Add the specified lock length.
+      $duration = $form_state['values']['choice'] == 'minutes' ? $form_state['values']['duration'] : 'indefinite';
+      islandora_object_lock_length_add_collection_lock($form_state['storage']['pid'], $duration);
     }
-    // Add the specified lock length.
-    islandora_object_lock_length_add_collection_lock($form_state['storage']['pid'], $form['duration']['#value']);
+    // If the value is being set back to default, remove the previous entry.
+    if ($collection_lock_length) {
+      if (($choice == 'minutes' && $form_state['values']['duration'] == $default_lock_length) || ($choice == 'indefinite' && $default_lock_length == FALSE)) {
+        islandora_object_lock_length_remove_collection_lock($form_state['storage']['pid'], $form_state['values']['duration']);
+      }
+    }
+    drupal_set_message(t('Collection level lock has been set.'));
   }
-  // If the value is being set back to default, remove the previous entry.
-  if (isset($collection_lock_length) && $form['duration']['#value'] == $default_lock_length) {
-    islandora_object_lock_length_remove_collection_lock($form_state['storage']['pid'], $form['duration']['#value']);
+  else {
+    drupal_set_message(t('The existing collection lock has been removed.'));
```

Comment:
Should... this be after the call to remove? Just thinking, in the case the call which does the removal fails, this could get confusing to debug.